WooCommerce/Classes/POS/Presentation/CardPresentPaymentAlert.swift

```diff
@@ -1,6 +1,24 @@
 import SwiftUI
 
 struct CardPresentPaymentAlert: View {
+    // TODO: Figure out whether this is the right choice vs @StateObject
+    // Using @StateObject resulted in the alert not being updated when
+    // alertViewModel changed – possibly we needed to invalidate the view's id.
+    @ObservedObject private var viewModel: CardPresentPaymentAlertSwiftUIViewModel
```

Comment:
Note that using StateObject meant that the alert shown on screen never changed, even when `showAlert` was sent...

Reply:
Good point, did you notice the alert content not changing during testing? It's interesting who is going to own this, and if there is no clear owner there might be unexpected behavior 🤔 Will keep this in mind if running into some unexpected issues down the line.

For view StateObject, it's possible to [specify an identity](https://developer.apple.com/documentation/swiftui/stateobject#Force-reinitialization-by-changing-view-identity) to force reinitialization. It isn't very straightforward to have an identity for the alert view model though.

Reply:
> did you notice the alert content not changing during testing?

Yep, for me it never changed. I removed my address from the store, but it never got to the "update your address" screen, it just sat on the initial state when the modal showed. I could see the states changing in the console, but the UI never updated to match them.